spec/requests/subject_spec.rb

```diff
@@ -231,113 +231,189 @@
       context "when requesting data for someone else" do
         it "renders the expected page" do
           expect(response).to render_template(:edit)
-          expect(response.body).to include("Upload their ID")
-          expect(response.body).to include("For example, a copy of their driving licence or passport")
-          expect(response.body).to include("For example an electricity or council tax bill in their name")
+          expect(response.body).to include("Upload their photo ID")
+          expect(response.body).to include("For example, a driving licence or passport")
         end
       end
 
       context "when requesting data for self" do
-        let(:information_request) { build(:information_request_for_self) }
+        let(:photo_upload) { create(:attachment) }
+        let(:information_request) { build(:information_request_by_friend, subject_photo_id: photo_upload.id) }
 
         it "renders the expected page" do
           expect(response).to render_template(:edit)
-          expect(response.body).to include("Upload your ID")
-          expect(response.body).to include("For example, a copy of your driving licence or passport")
-          expect(response.body).to include("For example an electricity or council tax bill in your name")
+          expect(response.body).to include("Upload their photo ID")
+          expect(response.body).to include("For example, a driving licence or passport. This can be a photograph, scan or photocopy of the original document. Maximum size: 7MB.")
         end
-      end
 
-      context "when submitting form with invalid data" do
-        it "renders page with error message" do
-          patch "/request", params: { request_form: { subject_photo: invalid_data, subject_proof_of_address: invalid_data } }
-          expect(response).to render_template(:edit)
-          expect(response.body).to include("There is a problem")
-          expect(response.body).to include("Add a file for Photo ID")
-          expect(response.body).to include("Add a file for Proof of address")
+        context "when submitting form with invalid data" do
+          it "renders page with error message" do
+            patch "/request", params: { request_form: { subject_photo: invalid_data } }
+            expect(response).to render_template(:edit)
+            expect(response.body).to include("There is a problem")
+            expect(response.body).to include("Add a file for Photo ID")
+          end
         end
-      end
 
-      context "when submitting form with valid data" do
-        it "goes to next step" do
-          patch "/request", params: { request_form: { subject_photo: valid_data, subject_proof_of_address: valid_data } }
-          expect(response).to redirect_to(next_step)
-        end
+        context "when returning to upload your proof of address page" do
```

Comment:
I can see this context has been renamed from "when returning to page", but it now refers to the proof of address which is wrong. It also has a bunch of tests included that were not there before. 

The purpose of this context was to test that once a file has been uploaded, the form can be resubmitted without having to upload another file.